vue/components/user-session/userprofile.vue

```diff
@@ -80,6 +94,138 @@ export default defineComponent({
     async changePassword() {
       await new ConsumerLogicApi('/account/change-password/').dialogForm('new');
     },
+    async handleSocialConnection(e: Event) {
+      const target = e.currentTarget as HTMLTextAreaElement;
+      if (!this.isSocialConnectionEnabled(target.id)) {
+        window.location.href = _.first(
+          _.filter(this.availableSocialConnections, (conn) => conn.name as string === target.id),
+        )?.url!;
+      }
+    },
+    mergeUsers(e: Event) {
+      e.preventDefault();
+      const user: String | null = (<HTMLInputElement>document.getElementById('merge-user-user')).value;
+      const passwd: String | null = (<HTMLInputElement>document.getElementById('merge-user-password')).value;
+      const account: Boolean | null = (<HTMLInputElement>document.getElementById('merge-user-account')).checked;
+      apiClient.post(
+        '/account/profile/merge-accounts',
+        { login: user, password: passwd, account },
+        { hideErrorNotice: true },
+      ).then(() => {
+        if (this.socialConnectionsModalPromise) {
+          dfModal.getDialogDefinition(this.socialConnectionsModalPromise)?.close();
+        }
+      }).catch((err) => {
+        if (err.response.status === HTTP_401_UNAUTHORIZED) {
+          if (this.socialConnectionsModalPromise) {
+            dfModal.getDialogDefinition(this.socialConnectionsModalPromise)?.close();
+          }
+          window.location.reload();
+          return;
+        }
+        showGeneralErrorNotification(err.response.data.detail);
+      });
+    },
+    async editSocialConnections() {
+      axios.all([
+        apiClient.get('account/social-auth-providers'),
+        apiClient.get('account/social-auth-providers-user'),
+      ]).then(axios.spread((available, used) => {
+        this.enabledSocialConnections = used.data;
+        this.availableSocialConnections = available.data;
+        if (_.size(available.data)) {
+          const socAccConfig = _.map(this.availableSocialConnections, (socAcc) => (
+            h('div', {
+              innerHTML: icons[socAcc.name as IconObjectKey],
+              style: this.getSocialConnectionStyle(socAcc.name),
+              onClick: this.handleSocialConnection,
+              id: socAcc.name,
+              'data-url': socAcc.url,
+            })));
+          this.socialConnectionsModalPromise = dfModal.message('', () => [
+            // eslint-disable-next-line vue/max-len
+            h('div', { style: 'display: flex; flex-direction: row; padding-top: 0.3em; padding-bottom: 1em; justify-content: space-around;' }, [
```

Comment:
uf: tu si naredil modal s template-om, samo da si template zrisal na roko. Je to optimalno?

Reply:
dodatno: Uroš trenutno dela na dynamicformsih, da bi obstajal en quick dialog form builder, kar si po mojem ti tukaj na roko naredil.